### ISSUE TITLE
Can you post the training code?

### ISSUE THREAD

**Issue by LiGuo12** [facebookresearch/segment-anything-2#115]
The forward function in the SAM2Base class is currently empty, and the loss function used is not specified. Could you please release the complete training code? This would allow the community to verify your results and perform fine-tuning. Your contributions and assistance are greatly appreciated.

**Comment by bhack** [comment]
It was already requested here for the prompts simulator https://github.com/facebookresearch/segment-anything-2/issues/114


and more in general here for the training/fine-tuning code https://github.com/facebookresearch/segment-anything-2/issues/12

**Comment by JunMa11** [comment]
Hi @LiGuo12 @bhack 
Here is the fine-tuning code for image segmentation
https://github.com/bowang-lab/MedSAM/blob/MedSAM2/README.md

**Comment by LiGuo12** [comment]
> Hi @LiGuo12 @bhack Here is the fine-tuning code for image segmentation https://github.com/bowang-lab/MedSAM/blob/MedSAM2/README.md

Thanks for the wonderful code. Looking forward to your fine-tuning weights on the medical datasets.

**Comment by LiGuo12** [comment]
> Hi @LiGuo12 @bhack Here is the fine-tuning code for image segmentation https://github.com/bowang-lab/MedSAM/blob/MedSAM2/README.md

Hi @JunMa11,
Thanks for the wonderful code. Looking forward to your fine-tuning weights on the medical dataset.

**Comment by bhack** [comment]
The most interesting and missing part is to fine-tune the decoder/prompt encoder on the video mode

**Comment by EricLina** [comment]
> The most interesting and missing part is to fine-tune the decoder/prompt encoder on the video mode

@ronghanghu

**Comment by haithamkhedr** [comment]
Thanks for you all for your interest. We released the training code and provide some examples to [get started](https://github.com/facebookresearch/segment-anything-2/tree/main/training#getting-started).

**Comment by bhack** [comment]
@haithamkhedr Thanks, what is the learnable decoder resolution at training time before the upscale? Can this be configured?
